hub/core/meta/encode/base_encoder.py

```diff
@@ -675,3 +675,9 @@ def _try_splitting_middle(
         self._encoded = np.concatenate((start, [new_row], end))
 
         return True
+
+    def _pop(self):
+        if self._encoded[-1][LAST_SEEN_INDEX_COLUMN] == 1:
```

Comment:
this should also not be == 1, but rather get the difference between last and second last row